clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -508,9 +508,30 @@ private void processAssignmentReceived(ConsumerGroupHeartbeatResponseData.Assign
      */
     private void replaceTargetAssignmentWithNewAssignment(
             ConsumerGroupHeartbeatResponseData.Assignment assignment) {
-        currentTargetAssignment.clear();
+
```

Comment:
I have to high level thoughts but I am not sure whether they are worth it:
1) Have you considered moving all the update logic into `LocalAssignment`? We could have a method such a `updateWith(Assignment)` which returns an `Optional` containing the new assignment if it was updated.
2) On a similar line, I wonder if we could have an `EMPTY` constant for the default `LocalAssignment(-1, null)` instead of relying on `null`. The reasoning of this one is that it avoids having to deal with `null` in a few places in this file.
What do you think?

Reply:
1. I did not do it, because the `LocalAssignment` is leaked from this file via the `currentAssignment` method, and I didn't necessarily want to put so much logic in the public interface `MembershipManager`. However, I think I could possibly define the return value of `currentAssignment` by a light interface, and then put the fat class with all the updating logic in here. I'll give it a try. 
2. I wouldn't necessarily call it `EMPTY` (to avoid confusion with an empty assignment), but rather `NONE` or something, but other than that it sounds like a good idea.